tools/linter/clang_tidy/run.py

```diff
@@ -352,11 +446,15 @@ def run(options: Any) -> int:
             if os.path.relpath(fname) != mapped_fname:
                 shutil.copyfile(fname, mapped_fname)
 
-    pwd = os.getcwd() + "/"
     if options.dry_run:
         print(clang_tidy_output)
-    for line in clang_tidy_output.splitlines():
-        if line.startswith(pwd):
-            print(line[len(pwd) :])
-
-    return len(warnings.keys())
+    elif returncode != 0:
+        print("[clang-tidy] Warnings detected!")
```

Comment:
What's the value of re-printing the warnings? At least on a small change it seems to mostly parrot what it's already said:


```
$ make clang-tidy CHANGED_ONLY=--changed-only                                                             (pytorch) [ 10:19:19 ]
x clang-tidy: Run clang-tidy
Warning detected in torch/csrc/utils/crash_handler.cpp
/data/users/davidriazati/pytorch/torch/csrc/utils/crash_handler.cpp:19:20: error: variable 'minidump_directory' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables,-warnings-as-errors]
static std::string minidump_directory; 
                   ^

Traceback (most recent call last):
  File "/home/davidriazati/local/miniconda3/envs/pytorch/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/davidriazati/local/miniconda3/envs/pytorch/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/data/users/davidriazati/pytorch/tools/linter/clang_tidy/__main__.py", line 194, in <module>
    main()
  File "/data/users/davidriazati/pytorch/tools/linter/clang_tidy/__main__.py", line 190, in main
    raise RuntimeError("Your code is not clang-tidy clean!")
RuntimeError: Your code is not clang-tidy clean!
[clang-tidy] Warnings detected!
Summary:
torch/csrc/utils/crash_handler.cpp:19:20: error: variable 'minidump_directory' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables,-warnings-as-errors]

```

Reply:
That's a good point. Perhaps I can aggregate the warnings and output a better summary. Something like
```
[warning-name]: <num-occurrences>
```

Chromium does this